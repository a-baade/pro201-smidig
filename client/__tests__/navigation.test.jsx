import React from "react";
import ReactDOM from "react-dom";
import { Link, MemoryRouter } from "react-router-dom";
import Navigation from "../components/navigation";
import Footer from "../components/footer";

describe("Navigation and Footer components", () => {
  it("should test navigation", function() {
    const divEl = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Navigation/>
      </MemoryRouter>,
      divEl
    );
    expect(divEl.innerHTML).toMatchSnapshot();
  });

  it("should test Link", function() {
    const link = document.createElement("Link");
    ReactDOM.render(
      <MemoryRouter>
        <Link to={"/about"} className={ "navLinks" }>About us</Link>
      </MemoryRouter>,
      link
    );
    expect(link.innerHTML).toMatchSnapshot();
  });

  it("should test footer", function() {
    const divEl = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Footer/>
      </MemoryRouter>,
      divEl
    );
    expect(divEl.innerHTML).toMatchSnapshot();
  });

})