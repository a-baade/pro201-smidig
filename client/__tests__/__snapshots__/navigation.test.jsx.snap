// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation and Footer components should test Link 1`] = `"<a class=\\"navLinks\\" href=\\"/about\\">About us</a>"`;

exports[`Navigation and Footer components should test footer 1`] = `"<footer><div class=\\"footerLinks\\"><a class=\\"footerLink\\" href=\\"/\\">Social Media</a><p>|</p><a class=\\"footerLink\\" href=\\"/\\">Donate</a><p>|</p><a class=\\"footerLink\\" href=\\"/\\">About us</a><p>|</p><a class=\\"footerLink\\" href=\\"/\\">Contact</a></div><div class=\\"footerSocial\\"><a href=\\"https://www.facebook.com/\\"><img src=\\"[object Object]\\" alt=\\"facebook\\"> </a><a href=\\"https://www.instagram.com/\\"><img src=\\"[object Object]\\" alt=\\"insta\\"></a><a href=\\"https://www.snapchat.com/\\"><img src=\\"[object Object]\\" alt=\\"snap\\"> </a><a href=\\"https://www.tiktok.com/\\"><img src=\\"[object Object]\\" alt=\\"tiktok\\"></a></div><div class=\\" footerCopyright \\"><p>Copyright Â© 2022 Melioria Impact. All rights reserved</p></div><div class=\\"footerSubscribe\\"><p>Subscribe here for news about our causes</p><form><input type=\\"email\\" placeholder=\\"example@email.com\\" value=\\"\\"><button>Subscribe</button></form></div></footer>"`;

exports[`Navigation and Footer components should test navigation 1`] = `"<nav><div><a href=\\"/\\"><img src=\\"[object Object]\\" class=\\"logo\\"></a><a class=\\"navLinks\\" href=\\"/charities\\">Charities</a><a class=\\"navLinks\\" href=\\"/about\\">About us</a><p class=\\"navLinks\\">Contact</p><p class=\\"navLinks\\">Other</p></div><div><div class=\\"search-box\\"><input class=\\"search-text\\" type=\\"text\\" placeholder=\\"Search Anything\\"><a href=\\"#\\" class=\\"search-btn\\"><img src=\\"[object Object]\\" class=\\"search\\"></a></div><div class=\\"profileDropdown\\"><img src=\\"[object Object]\\" class=\\"navIcons\\"><div class=\\"dropdownContent\\"><a class=\\"dropdownLink\\" href=\\"/login\\">Login</a><a class=\\"dropdownLink\\" href=\\"/\\">My dashboard</a></div></div></div></nav>"`;
